### PR TITLE
feat: Support other DID methods

### LINKED ISSUE #111: Support other DID methods
https://github.com/trustbloc/edge-agent/pull/108#discussion_r413393017

### PR DESCRIPTION
**Pull request by fqutishat**
closes #111

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: cmd/user-agent/src/pages/DIDManagement.vue [hunk 1]
```diff
@@ -80,10 +80,23 @@ SPDX-License-Identifier: Apache-2.0
...
+                                    <option value="Ed25519">Ed25519</option>
+                                </select>
+                                    <md-field style="margin-top: -15px">
+                                    </md-field>
+                                </div>
+                                <div class="md-layout-item md-size-100">
+                                    <md-icon>memory</md-icon> <select id="selectSignKey" v-model="selectSignKey" style="color: grey; width: 300px; height: 35px;">
+                                    <option value="">Select Signature Type</option>
                                     <option value="Ed25519Signature2018">Ed25519Signature2018</option>
-                                    <option value="JsonWebSignature2020">JsonWebSignature2020</option>
```
Comment (troyronda) [inline cmd/user-agent/src/pages/DIDManagement.vue:86]: Why is this being removed?

File: cmd/user-agent/src/pages/DIDManagement.vue [hunk 2]
```diff
@@ -80,10 +80,23 @@ SPDX-License-Identifier: Apache-2.0
...
-                                    <option value="" disabled="disabled">Select Signature Type</option>
+                                    <md-field maxlength="5">
+                                        <label class="md-helper-text">Type DID friendly name here</label>
+                                        <md-input v-model="anyDIDFriendlyName" id="anyDIDFriendlyName" required></md-input>
+                                    </md-field>
+                                </div>
+                                <div class="md-layout-item md-size-100">
+                                    <md-icon>memory</md-icon> <select id="privateKeyType" v-model="privateKeyType" style="color: grey; width: 200px; height: 35px;">
+                                    <option value="">Select Key Type</option>
+                                    <option value="Ed25519">Ed25519</option>
```
Comment (troyronda) [inline cmd/user-agent/src/pages/DIDManagement.vue:91]: Also need P256.

Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:91]: vendors DID don't support p256

Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:91]: when they support will add it

File: cmd/user-agent/src/pages/DIDManagement.vue [hunk 3]
```diff
@@ -141,20 +160,24 @@ SPDX-License-Identifier: Apache-2.0
...
 
                 const createDIDRequest = {
                     "publicKeys": [{
                         "id": keyset.keyID,
-                        "type": "JwsVerificationKey2020",
+                        "type": m.get(this.signType),
                         "value": keyset.publicKey,
                         "encoding": "Jwk",
                         "keyType": this.selectType,
-                        "usage": ["ops", "general"]
```
Comment (troyronda) [inline cmd/user-agent/src/pages/DIDManagement.vue:154]: which key will be used for ops?

File: cmd/user-agent/src/pages/DIDManagement.vue (hunk 1 above)
Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:86]: vendors DID don't support JsonWebSignature2020

Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:86]: when they support will add it

File: cmd/user-agent/src/pages/DIDManagement.vue (hunk 2 above)
Comment (troyronda) [reply cmd/user-agent/src/pages/DIDManagement.vue:91]: so this flow isn't used against TrustBloc DIDs?

File: cmd/user-agent/src/pages/DIDManagement.vue (hunk 3 above)
Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:154]: I will add another key for ops

File: cmd/user-agent/src/pages/DIDManagement.vue [hunk 4]
```diff
@@ -124,15 +136,22 @@ SPDX-License-Identifier: Apache-2.0
...
+                var m = new Map([["Ed25519Signature2018","Ed25519VerificationKey2018"], ["JsonWebSignature2020" ,"JwsVerificationKey2020" ]]);
+
                 this.errors.length = 0
                 if (this.friendlyName.length == 0) {
                     this.errors.push("friendly name required.")
                     return
                 }
                 if ((this.selectType == "")) {
-                    this.errors.push("key type required")
+                    this.errors.push("key type required1")
```
Comment (troyronda) [inline cmd/user-agent/src/pages/DIDManagement.vue:147]: required1?

Comment (sandrask) [inline cmd/user-agent/src/pages/DIDManagement.vue:147]: typo?

Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue]: done

Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue]: done

File: cmd/user-agent/src/pages/DIDManagement.vue (hunk 3 above)
Comment (fqutishat) [reply cmd/user-agent/src/pages/DIDManagement.vue:154]: done

**Pull request merged**